File: Factorial/factorial.cpp
#include <iostream>

using namespace std;

int factorial(int n)
{
    // Trường hợp cơ sở
    if (n == 0 || n == 1)
        return 1;

    // Trường hợp đệ quy
    return factorial(n - 1) * n;
}

int main()
{
    int num;
    cout << "Nhập số nguyên dương: ";
    cin >> num;

    int result = factorial(num);
    cout << num << "! = " << result;

    return 0;
}

File: Factorial/factorial.py
def factorial(n):
    # Trường hợp cơ sở
    if n == 0 or n == 1:
        return 1

    # Trường hợp đệ quy
    return factorial(n - 1) * n

if __name__ == '__main__':
    num = int(input('Nhập số nguyên dương: '))

    result = factorial(num)
    print(f'{num}! = {result}')